cmd/entrypoint/testutil_fake_hello_test.go

```diff
@@ -142,32 +142,55 @@ func FindCluster(envoyConfig *v3bootstrap.Bootstrap, predicate func(*v3cluster.C
 	return nil
 }
 
-func deltaSummary(t *testing.T, snap *snapshot.Snapshot) []string {
+func deltaSummary(t *testing.T, snaps ...*snapshot.Snapshot) []string {
 	summary := []string{}
 
 	var typestr string
 
-	for _, delta := range snap.Deltas {
-		switch delta.DeltaType {
-		case kates.ObjectAdd:
-			typestr = "add"
-		case kates.ObjectUpdate:
-			typestr = "update"
-		case kates.ObjectDelete:
-			typestr = "delete"
-		default:
-			// Bug because the programmer needs to add another case here.
-			t.Fatalf("missing case for DeltaType enum: %#v", delta)
+	for _, snap := range snaps {
+		for _, delta := range snap.Deltas {
+			switch delta.DeltaType {
+			case kates.ObjectAdd:
+				typestr = "add"
+			case kates.ObjectUpdate:
+				typestr = "update"
+			case kates.ObjectDelete:
+				typestr = "delete"
+			default:
+				// Bug because the programmer needs to add another case here.
+				t.Fatalf("missing case for DeltaType enum: %#v", delta)
+			}
+
+			summary = append(summary, fmt.Sprintf("%s %s %s", typestr, delta.Kind, delta.Name))
 		}
-
-		summary = append(summary, fmt.Sprintf("%s %s %s", typestr, delta.Kind, delta.Name))
 	}
 
 	sort.Strings(summary)
 
 	return summary
 }
 
+// getSnapshots is like f.GetSnapshot, but returns the list of every snapshot evaluated (rather than
+// discarding the snapshots from before predicate returns true).  This is particularly important if
+// you're looking at deltas; you don't want to discard any deltas just because two snapshots didn't
+// get coalesced.
```

Comment:
The thing that always bugged me about this test is that I feel like snapshot coalescing, in the `fake` world, should be deterministic. We should talk about that.

We should also consider making `f.GetSnapshot` coalesce deltas when skipping over snapshots... except, when you think about it, that's really dependent on ordering...

Reply:
I mostly feel like you're right, but I don't have enough of entrypoint in my head to discuss it in detail.

See also: The `watt` package in apro, and the "watchable" things in Telepresence.